config/src/main/java/com/typesafe/config/impl/AbstractConfigNode.java

```diff
@@ -0,0 +1,15 @@
+package com.typesafe.config.impl;
+
+import com.typesafe.config.ConfigNode;
+
+abstract class AbstractConfigNode implements ConfigNode {
+    final private Token token;
```

Comment:
it seems like the "responsibility" of this class is to be a node with only one token, so `ConfigNodeSingleToken` or something?

Some nodes will have multiple tokens right? Or not? I would expect the base class for all nodes to abstract that a little, maybe with something like `protected Iterable tokens()` and then for `ConfigNodeSingleToken` that would return `Collections.singletonList(token)` or the like.